test/test_utf8.cpp

```diff
@@ -169,6 +171,45 @@ TEST(UTF8_Compare_Core_utf8)
     CHECK_EQUAL(false, str_compare(u16sur2, u16sur2));
 }
 
+TEST(UTF8_Compare_Core_utf8_interned_strings)
```

Comment:
It would be better to not make a copy of the tests. I'd suggest modifying the existing tests to do the interned sort inside of the str_compare lambda and check that it is the same result as the non-interned comparison. The benefit of doing it that way, is that we can easily add more values to both cases, and we can directly see if there are any differences in the comparisons rather than doing a diff across tests.